### ISSUE TITLE
Remove ui-gmap-free-draw-polygons, and let ui-gmap-google-map handle drawing?

### ISSUE THREAD

**Issue by lukehorvat** [angular-ui/angular-google-maps#973]
Following on from #944 and #972, I'm starting to think (and this is only a suggestion) we should do away with the `ui-gmap-free-draw-polygons` directive. Instead, free-drawing could be defined via the `ui-gmap-google-map` directive.

For example, I'm imagining something like this:

``` html
<button ng-click="startDrawing()">Draw Polygon</button>

<ui-gmap-google-map center="map.center" zoom="map.zoom" draw="map.draw">
  <ui-gmap-polygon ng-repeat="polygon in map.polygons" path="polygon.path" fill="polygon.fill">
  </ui-gmap-polygon>
</ui-gmap-google-map>
```

``` javascript
$scope.map = {
  center: {
    latitude: -33.8,
    longitude: 151.2
  },
  zoom: 11,
  draw: null,
  polygons: []
};

$scope.startDrawing = function() {
  $scope.map.draw().then(function(path) {
    $scope.map.polygons.push({ path: path, fill: "#ff0000" });
  });
};
```

As you can see, `map.draw` is the same ol' function to call to enter drawing mode, but with one difference - the promise that it returns is resolved with the path of the polyline that was drawn by the user. Furthermore, when the user finishes drawing, no polygon is automatically rendered on the map. Instead, the onus is on you to do something with the polyline path. I think this is better, because it gives you full control over how drawn polygons are rendered (in the example above, I set the `fill`, which you can't do with the current `ui-gmap-free-draw-polygons` directive).

A map can either be drawn on or it can't, so I think it makes sense to define it at the `ui-gmap-google-map` level. What do you think?

**Comment by nmccready** [comment]
This makes sense at the same time your adding more responsibility to the map directive. To keep this cleaner it would need to be broken up into supportive factories and services.

**Comment by nmccready** [comment]
@cthrax thoughts? @zacronos ?

**Comment by cthrax** [comment]
Yeah, I think a service, or the controller paradigm would be more effective than adding it to map.
